### DISCUSSION TITLE
Export doesn't allow to use _space.parent?

### DISCUSSION THREAD

**Discussion by bakerwy** [fumitoh/modelx#129] [Q&A] [answered]
From the discussion #103,

https://github.com/fumitoh/modelx/discussions/103

using the answered method, I built RA model. 

But when I try to use this feature in **exported model**, it seems it is not working (I guess **export** doesn't support _space.parent)

Is there anyway I can use this method in exported model?

If not, is there any alternatives?

I hope I can get some clue.

Thanks.

**Comment by fumitoh** [comment, marked as answer]
You're right. Currently `_space.parent` doesn't work in the exported model, but `_space._parent` works. On the other hand, `_space._parent` does not work in the original model. I'll fix the issue by defining `_space._parent` also in modelx models.

For now, a workaround is to define `is_exported` in `RA` as below.

```python
import modelx as mx

m = mx.new_model()
RA = m.new_space("RA")

# Set the allow_none property to True so that all formulas in RA can return None.
RA.allow_none = True

# Parameterize Space RA with `calc_loop`.
# This means RA creates exact copies of itself parameterized by calc_loop on the fly, such as RA[0], RA[1], RA[2], etc.
# They are dynamic child spaces of RA.
RA.parameters = ("calc_loop",)

# Define BEL_LAPSE in RA. In the formula, calc_loop is the value given to the parameter of RA[calc_loop].
# For example, calc_loop == 0 in RA[0].BEL_LAPSE(), calc_loop == 1 in RA[1].BEL_LAPSE(), and so on.
# In the formula, `_space` is a special name that represents the parent space of the Cells.
# For example, _space is RA[2] in RA[2].BEL_LAPSE(). _space.parent represents the parent space of RA[2], which is RA.
# So, _space.parent[1].BEL_LAPSE() means RA[1].BEL_LAPSE()

RA.is_exported = False

@mx.defcells
def BEL_LAPSE():
    if calc_loop == 0:
        return None
    elif calc_loop == 1:
        return 120
    else:
        if is_exported:
            return _space._parent[1].BEL_LAPSE()
        else:
            return _space.parent[1].BEL_LAPSE()

for i in range(4):
    print(f"calc_loop: {i}, BEL_LAPSE: {RA[i].BEL_LAPSE()}")

m.export("temp_nomx")
```
Then for the expoted model, you can manually set `is_exported` to `True`:

```python

from temp_nomx import Model1

Model1.RA.is_exported = True

for i in range(4):
    print(f"calc_loop: {i}, BEL_LAPSE: {Model1.RA[i].BEL_LAPSE()}")

```

**Comment by bakerwy** [comment]
Thanks alot! 

BTW,
I tried _space[1].BEL_LAPSE() and it is working in exported model. How is it working? and if I use _space[1].BEL_LAPSE(), instead of _space._parent[1], is there any potential problem?

**Reply by fumitoh** [reply]
`_space[1].BEL_LAPSE() ` in `RA[2].BEL_LAPSE()` translates to `RA[2][1].BEL_LAPSE() `  instead of  `RA[1].BEL_LAPSE() ` ,
so you're creating `RA[2][1]`. 
In below,  `__Space3[1]` and `__Space4[1]` are `RA[2][1]` and `RA[3][1]`  respectively, which you didn't intend to create.

![RA](https://github.com/fumitoh/modelx/assets/8450892/e7c0f493-f6b6-4e5b-93ed-691851e3cd12)

**Reply by bakerwy** [reply]
Ah.. thanks a lot!

**Comment by fumitoh** [comment]
[modelx v0.26.0](https://docs.modelx.io/en/latest/releases/relnotes_v0_26_0.html) has been released, and now `_space._parent` is available in both modelx models and exported models.
